test_gpios.py

```diff
@@ -17,7 +17,23 @@ async def asyncSetUp(self):
             refresh_interval=0,
             dial_options=DialOptions(credentials=conf.creds)
         )
-        self.robot = await RobotClient.at_address(conf.address, opts)
+
+        # As of September 2023, there is a race condition in the Python SDK,
+        # and some connections time out if they get opened and closed too
+        # quickly. Try a few times before giving up.
+        # TODO[RSDK-4846]: Simplify this when the bug is fixed.
```

Comment:
Might be useful to link the ticket to the blocking ticket so you get notified when the blocking ticket is resolved.

Reply:
Link which ticket to it? RSDK-4766 will be closed as soon as I'm happy that the Beaglebone's tests are passing, even if RSDK-4846 is still unresolved, and that's the last board-canary-related ticket I know of. but I reported RSDK-4846, so hopefully I'll remember to undo this when it's ready.

I can comment on RSDK-4846 with a reminder to revert this PR when that happens, though.